### PR TITLE
Issue#506 organization editable

### PR DESCRIPTION
**Pull request by ioaNikas**
- [x] Organization is editable (only address at the moment, but can be extended to other fields later) using the new editable side-nav component.
- [x] Rename and sort components and pages for easier reading.
- [x] Invitations are triggered once, and queried by organizationId. (issue#645)

### CODE REVIEW

File: libs/notification/invitation/+state/invitation.service.ts [hunk 1]
```diff
@@ -19,16 +19,16 @@ export class InvitationService {
   public get userInvitations$() {
     return this.authQuery.user$.pipe(
       filter(user => !!user),
-      switchMap(user => this.db.fromQuery(this.getInvitationByUserId(user.uid))),
-      tap((invitations: any) => this.store.set(invitations)) // TODO : Find a way to cast invitations as Invitation[];
+      switchMap(user => this.db.fromQuery(this.getInvitationsByOrgId(user.orgId))),
+      tap((invitations: any) => this.store.set(invitations))
```
Comment (GrandSchtroumpf) [inline libs/notification/invitation/+state/invitation.service.ts]: You should type "invitiations"

File: libs/organization/src/lib/components/organization-display/organization-display.component.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<ng-container *ngIf="organization; else: loading">
+  <mat-card class="mat-elevation-z1">
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-display/organization-display.component.html]: Don't specify the elevation everywhere. If there is a "standard" elevation level it should be settled in the theme.

Comment (GrandSchtroumpf) [reply libs/organization/src/lib/components/organization-display/organization-display.component.html]: Also, I think the `fxLayout="row"` should be on the mat-card, no ? 
```html
<mat-card fxLayout="row">
  <img />
  <mat-card-content></>
</mat-card>
```

File: libs/organization/src/lib/components/organization-display/organization-display.component.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <!-- TODO: issue#579 import picture -->
+      <img
+        mat-card-avatar
+        src="https://material.angular.io/assets/img/examples/shiba2.jpg"
+        alt="Profile picture"
+      />
+      <article class="information" fxLayout="column" fxLayoutAlign="start start">
+        <div>
+          <h4 class="name">
+            <b>{{ organization.name }}</b>
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-display/organization-display.component.html]: Remove b tag, it's already a title. Use material typpography

File: libs/organization/src/lib/components/organization-form/organization-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Component, ChangeDetectionStrategy } from '@angular/core';
+import { ControlContainer } from '@angular/forms';
+
+@Component({
+  selector: 'organization-form',
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-form/organization-form.component.ts]: Add `[formGroupName] organization-form, [formGroup] organization-form` to force the selector to have `formGroupName` or `formGroup` attribute.

File: libs/organization/src/lib/components/organization-form/organization-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './organization-form.component.html',
+  styleUrls: ['./organization-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class OrganizationFormComponent {
+  constructor(public controlContainer: ControlContainer) {}
+
+  public get control() {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/components/organization-form/organization-form.component.ts]: if there is an OrganizationForm you can do : `public get control(): OrganizationForm {}`

File: libs/organization/src/lib/organization.module.ts [hunk 6]
```diff
@@ -21,35 +21,39 @@ import { MatMenuModule } from '@angular/material/menu';
...
-    component: OrgFormComponent
+    path: 'create',
+    component: OrganizationCreateComponent
   },
   {
     path: ':orgId',
     children: [
       { path: '', redirectTo: 'view', pathMatch: 'full' },
-      { path: 'view', component: OrgViewComponent },
+      { path: 'view', component: OrganizationEditableComponent },
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization.module.ts]: It's suppose to be "edit" the route now.

File: libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    private query: OrganizationQuery,
+    private permissionsQuery: PermissionsQuery,
+    private service: OrganizationService,
+    private snackBar: MatSnackBar
+    ) {}
+
+  ngOnInit() {
+    this.organization$ = this.query.select('org');
+    this.isSuperAdmin$ = this.permissionsQuery.isSuperAdmin$;
+    this.organizationProfileForm = new OrganizationProfileForm(this.query.getValue().org);
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts]: You need to keep listening of the update of the server I think : 
```typescript
this.organizationProfileForm = new OrganizationProfileForm();
this.organization$ = this.query.select('org').pipe(tap(org => this.organizationProfileForm.patchValue(org));
this.isSuperAdmin$ = this.permissionsQuery.isSuperAdmin$;
```

Comment (GrandSchtroumpf) [reply libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts]: Maybe the best would be to have : 
```typescript
this.organization$ = this.query.select('org').pipe(
  tap(org => this.organizationProfileForm.patchValue(org)),
  switchMap(_ => this.organizationProfileForm.valueChanges)
);
```
Like that your form is always in sync with the DB, but still you have local changes available. But the effect might be weird if someone is updating the DB in the same time as you're writing...

File: libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  public get organizationInformations$() {
+    return this.organizationProfileForm.valueChanges.pipe(startWith(this.organizationProfileForm.value));
+  }
+
+  public openSidenav() {
+    this.opened = true;
+  }
+
+  public update() {
+    try {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts]: All this should be a generic method that should be reuse all the time. Let me check of to do that.

File: libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts (hunk 7 above)
Comment (ioaNikas) [reply libs/organization/src/lib/pages/organization-editable/organization-editable.component.ts]: I really don't get what you're trying to explain me. Maybe we can have a short call about this to make it clear for me ?

**Pull request merged**